tools/api-markdown-documenter/src/LintApiModel.ts

```diff
@@ -0,0 +1,308 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	ApiDocumentedItem,
+	type ApiItem,
+	ApiItemContainerMixin,
+	type ApiModel,
+} from "@microsoft/api-extractor-model";
+import type { DocInheritDocTag } from "@microsoft/tsdoc";
+
+import { defaultLoadModelOptions, loadModel, type LoadModelOptions } from "./LoadModel.js";
+import { noopLogger } from "./Logging.js";
+import { resolveSymbolicReference } from "./utilities/index.js";
+import { fail } from "node:assert";
+
+/**
+ * Linter check options.
+ *
+ * @public
+ */
+export interface LinterOptions {
+	/**
+	 * Whether or not to evaluate `{@link}` and `{@inheritDoc}` references as a part of the linting process.
+	 *
+	 * @remarks
+	 * Note: these checks do not evaluate the *syntax* of reference tags.
+	 * API-Extractor does that as a part of its model generation process.
+	 * This option is for verifying that symbolic references point to valid targets within the API model..
+	 *
+	 * @defaultValue `true`
+	 */
+	checkReferences?: boolean;
+}
+
+/**
+ * {@link lintApiModel} options.
+ */
+export interface LintApiModelOptions extends LoadModelOptions, LinterOptions {}
+
+/**
+ * {@link LintApiModelOptions} defaults.
+ */
+const defaultLintApiModelOptions: Required<Omit<LintApiModelOptions, "modelDirectoryPath">> = {
+	...defaultLoadModelOptions,
+	checkReferences: true,
+};
+
+/**
+ * Kinds of linter errors reported by {@link lintApiModel}.
+ */
+export enum LinterErrorKind {
```

Comment:
I haven't used them enough to remember their pros/cons in detail, but would `const enum` be better here?